File: Network.m
classdef Network
   properties
      data_folder
      fsp
      vsp
      demands
      tanks
   end
   
   methods
      function obj = init(obj, data_folder)
        obj.data_folder = data_folder;
        obj.fsp = obj.read_data_from_file(fullfile(data_folder, 'fsp.csv'));
        obj.fsp.facility = string(obj.fsp.facility);
        
        obj.tanks = obj.read_data_from_file(fullfile(data_folder, 'tanks.csv'));
        obj.tanks.demand = string(obj.tanks.demand);
        
      end

      function data = read_data_from_file(obj, file_path)
        data = readtable(file_path);
        
      end

   end

   methods(Static)
       function mat = get_cumulative_mat(T, multiplier)
           % multiplier can be flow direction 1 or -1
           mat = tril(ones(T, T)) * multiplier;
       end

   end
end

File: network.py
import os
import pandas as pd
import numpy as np


class Network:
    def __init__(self, data_folder):
        self.data_folder = data_folder

        self.fsp = pd.read_csv(os.path.join(self.data_folder, 'fsp.csv'))
        self.vsp = pd.read_csv(os.path.join(self.data_folder, 'vsp.csv'))
        self.tanks = pd.read_csv(os.path.join(self.data_folder, 'tanks.csv'), index_col='tank')

        self.n_fsp = len(self.fsp)
        self.n_vsp = len(self.vsp)
        self.n_tanks = len(self.tanks)

        self.fsp_names = self.fsp.loc[:, 'name']
        self.vsp_names = self.vsp.loc[:, 'name']

    @staticmethod
    def get_cumulative_mat(n, multiplier):
        """ multiplier can be flow direction 1 or -1 """
        return np.tril(np.ones((n, n))) * multiplier